feat(doc): Add index & filename guides (#11)
getsentry/frontend-handbook: README.md

The frontend codebase is currently located under `src/sentry/static/sentry/app` in sentry and `static/getsentry` in getsentry. (We intend to align to `static/sentry` in future.)

## Folder & File structure

### File Naming

> - Name a file meaningfully, based on what they are or do

Reviewer: ```suggestion
- Name a file meaningfully, based on the how the module's functions, or classes are used or the application section they are used in.
```

> - If not necessary, don't use prefixes or sufixes (ie, dataScrubbingEditModal, dataScrubbingAddModal)

Reviewer: ```suggestion
- Unless necessary, don't use prefixes or suffixes (ie. `dataScrubbingEditModal`, `dataScrubbingAddModal`) instead favor names like `dataScrubbing/editModal`.
```

Including an example of what is good could help here.

Author: sure :)

### Using `index.(j|t)?(sx)`

> Having an `index` file in a folder provides a way to implicitly import the main file without specifying it

The use of an index file should comply with the following rules:

> - If the folder is created to group components that are used together, and there is a "master" component, I.e some component that uses all the components within the grouping (examples, avatar, idBadge). The master component should be the index file.

Reviewer: ```suggestion
- If the folder is created to group components that are used together, and there is an entrypoint component, that uses the components within the grouping (examples, avatar, idBadge). The entrypoint component should be the index file.
```

> - There must be NO `index.(j|t)?(sx)` file if the folder contains components used in other parts of the app regardless of the master file. (ie, actionCreators, panels)

Reviewer: ```suggestion
- *Don't* use an `index.(j|t)?(sx)` file if the folder contains components used in other parts of the app regardless of the entrypoint file. (ie, actionCreators, panels)
```

> - DO NOT use the index file just to re-export. Prefer importing individual components instead.

Reviewer: ```suggestion
- *Don't* use an index file just to re-export. Prefer importing individual components instead.
```

I think these make the recommendations clearer 🤷
